### PR TITLE
bugfix/RHT-4505-update-response-error-message-from-logon

### PR DESCRIPTION
**Pull request by ntqdinh-axonivy**

### CODE REVIEW

**Comment by ntqdinh-axonivy** [comment]
Hi @ivy-rew 
Our customer project faces an error of 422: Unprocessable Entity in the step logon into DocuWare.

image

We tried to send that request from Postman with the customer's configuration but could not reproduce it. 
So, I created this PR to get an error message (since DocuWare uses 422 code for many scenarios e.g. The expression type ... is not recognized or Fill out the required field... ) and detect the root cause of the logon failure.

File: docuware-connector/src/com/axonivy/connector/docuware/connector/DocuWareAuthFeature.java [hunk 1]
```diff
@@ -64,9 +66,8 @@ public void filter(ClientRequestContext reqContext) throws IOException {
       Configuration configuration = reqContext.getConfiguration();
       FeatureConfig config = new FeatureConfig(configuration, DocuWareAuthFeature.class);
       Form form = new Form().param("UserName", config.readMandatory(Property.USERNAME))
-        .param("Password", config.readMandatory(Property.PASSWORD))
-        .param("HostID", config.readMandatory(Property.HOSTID))
-        .param("RedirectToMyselfInCaseOfError", "false");
+          .param("Password", config.readMandatory(Property.PASSWORD))
+          .param("HostID", config.readMandatory(Property.HOSTID)).param("RedirectToMyselfInCaseOfError", "false");
```
Comment (ivy-rew) [inline docuware-connector/src/com/axonivy/connector/docuware/connector/DocuWareAuthFeature.java:70]: please undo these formatting changes ... the code was more readable before

File: docuware-connector/src/com/axonivy/connector/docuware/connector/DocuWareAuthFeature.java [hunk 2]
```diff
@@ -95,16 +96,15 @@ private String getLogonUrl(ClientRequestContext reqContext, FeatureConfig config
       return logon.toString();
     } catch (Throwable t) {
       String message = String.format(
-              "Could not determine DocuWare target URL automatically, please set it in REST client property '%s'. Put there the same URL as used for the client.",
-              Property.LOGONURL);
+          "Could not determine DocuWare target URL automatically, please set it in REST client property '%s'. Put there the same URL as used for the client.",
+          Property.LOGONURL);
```
Comment (ivy-rew) [inline docuware-connector/src/com/axonivy/connector/docuware/connector/DocuWareAuthFeature.java:100]: undo

File: docuware-connector/src/com/axonivy/connector/docuware/connector/DocuWareAuthFeature.java [hunk 3]
```diff
@@ -95,16 +96,15 @@ private String getLogonUrl(ClientRequestContext reqContext, FeatureConfig config
...
       throw new IllegalStateException(message, t);
     }
   }
 
   protected Response logon(ClientRequestContext reqContext, Form form, String target) {
-    Response response = reqContext.getClient().target(target)
-            .request(MediaType.APPLICATION_FORM_URLENCODED_TYPE)
-            .accept(MediaType.APPLICATION_JSON).post(Entity.form(form));
+    Response response = reqContext.getClient().target(target).request(MediaType.APPLICATION_FORM_URLENCODED_TYPE)
+        .accept(MediaType.APPLICATION_JSON).post(Entity.form(form));
```
Comment (ivy-rew) [inline docuware-connector/src/com/axonivy/connector/docuware/connector/DocuWareAuthFeature.java:107]: undo

File: docuware-connector/src/com/axonivy/connector/docuware/connector/utils/RequestResponseUtils.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+      try {
+        while ((line = reader.readLine()) != null) {
+          stringBuilder.append(line);
+        }
+      } catch (IOException e) {
+        Ivy.log().warn(e.getMessage());
+      }
+      message = stringBuilder.toString();
+    }
+    return message;
```
Comment (ivy-rew) [inline docuware-connector/src/com/axonivy/connector/docuware/connector/utils/RequestResponseUtils.java:29]: IOUtils.toString(inputStream, StandardCharsets.UTF_8);

**Review by ivy-rew — CHANGES_REQUESTED** [review]
a good idea; to log the entity on error. please do:
- not re-format code for no reason.
- simplify the IO code by using appache-commons

**Comment by ntqdinh-axonivy** [comment]
Hi @ivy-rew 
Thanks for your hint, it makes the code pretty cleaner. I reset the code base to the commit before I started to work so this commit is automatically closed. :( 
I have adapted the new code with your suggestion in the below PR already. Once again, thanks for your support.
 https://github.com/axonivy-market/docuware-connector/pull/49

**Pull request closed without merging**